src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessonListing.test.tsx

```diff
@@ -93,4 +108,30 @@ describe("Lesson listing page", () => {
       ).rejects.toThrowError("no context.params");
     });
   });
+  describe("tracking", () => {
+    test("It calls tracking.lessonSelected with correct props when clicked", async () => {
+      const { getByText } = render(
+        <LessonListPage curriculumData={lessonListingFixture()} />,
+      );
+
+      const lesson = getByText("Add two surds");
+
+      act(() => {
+        lesson.click();
```

Comment:
I'd recommend using `userEvent` to handle clicks wherever they're being tested to get as close to 'real' behaviour as possible - it also wraps in act so you shouldn't need to do it manually

Reply:
Yeah I originally tried this but it wasn't calling the mocked tracking function, manually checked and avo was sending events. Maybe we could take a look at this together?

Reply:
oh good to know! that't a bit troubling though - were you `await` ing the userEvent.click?

Reply:
```
     act(async () => {
        await userEvent.click(lesson);
      });
```
I was using act - which failed the test
This worked  `  await userEvent.click(lesson);`

Reply:
```
    await act(async () => {
      await userEvent.click(unit);
    });
```
It was await! - was missing from the start of act on my previous attempt, this passes and removes act warning from tests.